src/program/db/db_functions.py

```diff
@@ -490,16 +428,18 @@ def _run_thread_with_db_item(fn, service, program, input_item: "MediaItem" = Non
                 for i in fn(input_item):
                     if isinstance(i, (MediaItem)):
                         with db.Session() as session:
-                            _check_for_and_run_insertion_required(session, i)
+                            store_or_update_item(i)
                     yield i
         return
     else:
         # Content services
         for i in fn():
             if isinstance(i, (MediaItem)):
+                store_or_update_item(i)

```

Comment:
We cant just blindly add the item, this will result in duplicates. Look through the database for existing items with the same imdb_id first and if found then merge in this method.